test_runner/src/main/kotlin/ftl/args/CreateIosArgs.kt

```diff
@@ -0,0 +1,19 @@
+package ftl.args
+
+import ftl.config.IosConfig
+import ftl.config.ios.IosFlankConfig
+import ftl.config.ios.IosGcloudConfig
+
+fun createIosArgs(
+    config: IosConfig? = null,
+    gcloud: IosGcloudConfig = config!!.platform.gcloud,
+    flank: IosFlankConfig = config!!.platform.flank,
+    commonArgs: CommonArgs = config!!.common.createCommonArgs(config.data)
```

Comment:
this is a little bit unclear for me, `config` is nullable, but later on we declare that it is not null. So probably IosConfig is not nullable type.

Reply:
Right, it doesn't looks well and should be explained, `config` is nullable cause when `gcloud`, `flank`, `commonArgs` are set explicit the `config` is redundant from logical perspective. Possible solution to get rid of null is:
```
fun createIosArgs(
    config: IosConfig
) = createIosArgs(
    gcloud = config.platform.gcloud,
    flank = config.platform.flank,
    commonArgs = config.common.createCommonArgs(config.data)
)

private fun createIosArgs(
    gcloud: IosGcloudConfig,
    flank: IosFlankConfig,
    commonArgs: CommonArgs
) = IosArgs(
    commonArgs = commonArgs,
    xctestrunZip = gcloud.test!!.processFilePath("from test"),
    xctestrunFile = gcloud.xctestrunFile!!.processFilePath("from xctestrun-file"),
    xcodeVersion = gcloud.xcodeVersion,
    devices = gcloud.device!!,
    testTargets = flank.testTargets!!.filterNotNull()
)
```
But it gives 2 functions instead of 1 and more lines. I'm am not sure which is better.